views.go

```diff
@@ -201,43 +233,45 @@ func (lk *logKeeper) createTest(w http.ResponseWriter, r *http.Request) {
 		lk.render.WriteJSON(w, http.StatusNotFound, apiError{Err: "build not found"})
 		return
 	}
+	if build.S3 {
+		exists, err := lk.opts.Bucket.CheckMetadata(r.Context(), buildID, "")
```

Comment:
This explicitly makes the behavior that if a build has "s3=true" but doesn't have the appropriate build metadata, we avoid doing the S3 upload/creation _and_ the DB one as well. Is that the intent here? One could also put this into the S3 parallel goroutine but it sounds like we want to just fail entirely at this point if the metadata isn't there?

Reply:
My assumption was that we would want to fail here if `s3=true` and the build does not exist in S3 since it would be invalid to have a test without a build. I did not put it in the goroutine because the code starts to get more messy than it already is since in some cases we want to return a 404 and others a 500.

Reply:
Ok. Wanted to confirm